packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -182,28 +183,33 @@ export class PendingStateManager implements IDisposable {
 	public async applyStashedOpsAt(seqNum?: number) {
 		// apply stashed ops at sequence number
 		while (!this.initialMessages.isEmpty()) {
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			const nextMessage = this.initialMessages.peekFront()!;
 			if (seqNum !== undefined) {
-				if (nextMessage.referenceSequenceNumber > seqNum) {
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				const peekMessage = this.initialMessages.peekFront()!;
+				if (peekMessage.referenceSequenceNumber > seqNum) {
 					break; // nothing left to do at this sequence number
 				}
-				if (nextMessage.referenceSequenceNumber < seqNum) {
+				if (peekMessage.referenceSequenceNumber < seqNum) {
 					throw new Error("loaded from snapshot too recent to apply stashed ops");
 				}
 			}
-
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const nextMessage = this.initialMessages.shift()!;
 			try {
 				// applyStashedOp will cause the DDS to behave as if it has sent the op but not actually send it
 				const localOpMetadata = await this.stateHandler.applyStashedOp(nextMessage.content);
-				nextMessage.localOpMetadata = localOpMetadata;
+				if (!this.stateHandler.isAttached()) {
+					if (localOpMetadata !== undefined) {
```

Comment:
if we are not attached, then there shouldn't be any local op metadata, and we don't track pending messages, as there are not messages when not attached.